### ISSUE TITLE
Any "Getting Started" documentation?

### ISSUE THREAD

**Issue by tonydemark** [CakePHP-Copula/Linkedin#2]
Is there any "Getting Started" documentation to show how to pull basic information via this plugin? Something simple like getting a profile into a variable and setting it for display?

I've never worked with Plugins before and am getting a host of errors on install, no matter what I try. They seem to be very basic errors (eg, "Datasource class Apis.Apis could not be found."), but I've set the system up exactly as the front page says.

**Comment by ProLoser** [comment]
That error means you forgot to install the api plugin too.

**Comment by tonydemark** [comment]
That's the funny thing ... the api plugin is there but it seems like it is not.

htdocs/app/Plugin:
- apis/
- http_socket_oauth/
- linkedin/

I also tried it in htdocs/plugins and tried camel casing the directories.

In htdocs/Config/bootstrap.php I've tried CakePlugin::loadAll() and load()ing individually.

This seems like something incredibly silly I am missing, but frustratingly can't find.

**Comment by ProLoser** [comment]
If you are on cake2 the folders must be camel cased not underscores
On Jan 20, 2012 9:30 AM, "shadow349" <
reply@reply.github.com>
wrote:

**Comment by tonydemark** [comment]
Tried it as:

htdocs/app/Plugin:
- Apis/
- HttpSocketOauth/
- Linkedin/

and still no joy. It's gotta be something simple I'm doing wrong, right?

**Comment by ProLoser** [comment]
I reviewed the documentation and updated. Cake2 no longer uses driver in the database.php file. Make sure you have those changes reflected. Looks like you're using Plugin (singular) which is correct too.

**Comment by ProLoser** [comment]
I'm probably going to add support for a 'section' field so that you can leverage the API url detection used by other datasources. In other words, the config file which maps out the LinkedIn api. My documentation still works fine (I have recently tested and confirmed this on my own code).

If you need to perform write commands (delete/update/save) I haven't gotten that far with that in my datasources yet as they provide new problems. I may make it so that you must pass also add a field to the data to be saved called 'section' so that I may properly figure out what you are trying to save. The logic for differentiating between create and update is a little more complex (since they both use $model->save()) and can either be handled by ALSO specifying this in a save field OR some really heavy code which actually checks the server for the records existence (something i don't feel like doing). Wish I could directly access DS::update() from the model :/

**Comment by jamesduncancss** [comment]
This looks like a really great plugin and perfect for my needs - (and maybe this is out of scope for the plugin itself) - but it would be really cool if there was a touch of some "Hello World" type documentation - admittedly I'm pretty new to CakePHP so understandable if I need to look elsewhere.

But I've installed things as described and now am at a bit of a dead end. I'll stumbling around various dev sources to try and piece this all together - do I now use the LinkedIn code from their dev site in my views?

Thanks

**Comment by ProLoser** [comment]
Did you read the instruction on the API datasource page? 

They provide examples of how to authenticate and then query for data. Using the data in the views is the same as if you queried the database. 

jamesduncancss reply@reply.github.com wrote:

This looks like a really great plugin and perfect for my needs - (and maybe this is out of scope for the plugin itself) - but it would be really cool if there was a touch of some "Hello World" type documentation - admittedly I'm pretty new to CakePHP so understandable if I need to look elsewhere.

But I've installed things as described and now am at a bit of a dead end. I'll stumbling around various dev sources to try and piece this all together - do I know use LinkedIn code from their dev site in my views?

Thanks

---

**Comment by jamesduncancss** [comment]
Yes - I'm going through it now - hence my comment (and maybe this is out of scope for the plugin itself) 

Anyways - still trying to piece this together with little luck - will soldier on.

**Comment by ProLoser** [comment]
Let me know if you get stuck and feel encouraged to update the documentation as I know it's small. 

Mainly you need to use the component to authenticate, and then attach the datasource to any model you want and treat it as you normally would. Just remember that any action you query from must have the component loaded.

jamesduncancss reply@reply.github.com wrote:

Yes - I'm going through it now - hence my comment (and maybe this is out of scope for the plugin itself) 

Anyways - still trying to piece this together with little luck - will soldier on.

---

**Comment by jamesduncancss** [comment]
Thanks, that info helps a lot - much appreciated!

re: documentention - Ok, I'll try for sure - I'm getting there and of course this is a great bit of code/work - so I'll try to send along some updates once I have it pieced together - thanks again for getting back so promptly

**Comment by ProLoser** [comment]
If you're on irc.freenode.net you can ping me in the #cakephp channel or just send me a pm: ProLoser|work
